### PR TITLE
Fix issue 1465: fix jit in example

### LINKED ISSUE #1465: advi.py example is broken with jit
advi.py example is broken with jit: @partial(jit, static_argnums=(0, 1, 2, 4))
If jit is removed, then It works.

### PR DESCRIPTION
**Pull request by xysun**
fixes #1465

### CODE REVIEW

File: examples/advi.py [hunk 1]
```diff
@@ -55,7 +55,7 @@ def batch_elbo(logprob, rng, params, num_samples):
 
 # ========= Helper function for plotting. =========
 
-@partial(jit, static_argnums=(0, 1, 2, 4))
+@partial(jit, static_argnums=(0, 1, 2))
 def mesh_eval(func, x_limits, y_limits, params, num_ticks=101):
```
Comment (xysun) [inline examples/advi.py]: The 4th positional argument, `num_ticks`, has a default value, and is never explicitly supplied during invocation. Hence the error. 

I took a look at [f_jitted]() but don't see a way to handle this scenario (i.e., include the default args)

Comment (xysun) [reply examples/advi.py]: Oh hold on I may be able to do this

**Comment by xysun** [comment]
An alternative approach is to update the check in [`jit`](https://github.com/google/jax/blob/master/jax/api.py#L135) function to:

```
if static_argnums and max(static_argnums) >= len(args) + len(fun.__defaults__):
```

The downside is `fun.__defaults__` is type level and does not consider whether the default argument has been supplied or not. This may let the actually wrong `static_argnums` pass the check, eg. consider this function:
```
@jit(static_nums=(0,1,2) # wrong
def f(a,b=2):
  pass
```

An invocation of `f(1,2)` will pass this check since `2 < 2 + 1`, although one could argue nothing would break (the `dyn_argnums` compute won't be impacted)

**Comment by xysun** [comment]
@mattjj @shoyer Can you take a look please? Am curious your thoughts on we could handle `static_argnums` check for default arguments.

**Comment by hawkinsp** [comment]
I changed this PR to use a wrapper function to avoid applying `static_argnums` to a keyword argument. This should make everything work until we add keyword static argument support.

**Pull request merged**